pkgs/by-name/sh/shopify-cli/package.nix

```diff
@@ -1,31 +1,79 @@
 {
-  buildNpmPackage,
   lib,
-  testers,
-  shopify-cli,
+  stdenv,
+  fetchFromGitHub,
+  pnpm,
+  faketty,
+  nodejs,
+  versionCheckHook,
+  makeBinaryWrapper,
+  nix-update-script,
 }:
 let
   version = "3.86.1";
 in
-buildNpmPackage {
+stdenv.mkDerivation (finalAttrs: {
   pname = "shopify";
-  version = version;
-
-  src = ./manifests;
+  version = "3.86.1";
 
-  npmDepsHash = "sha256-GAE4zfSqk7oM2ecojPC4REFGRdRwvqCnA9L2L7LMIfQ=";
-  dontNpmBuild = true;
+  src = fetchFromGitHub {
+    owner = "shopify";
+    repo = "cli";
+    tag = finalAttrs.version;
+    hash = "sha256-wEddzW5/+qdtNTxdUs7YEA5vk6/KjrVOgWvIeo0o2ww=";
+  };
 
-  passthru = {
-    updateScript = ./update.sh;
-    tests.version = testers.testVersion {
-      package = shopify-cli;
-      command = "shopify version";
-    };
+  pnpmDeps = pnpm.fetchDeps {
+    inherit (finalAttrs) pname version src;
+    fetcherVersion = 2;
+    hash = "sha256-JhyZpkrp78FECH6UKYYuhWF2w/mYW1BQG5FIsWh5GRE=";
   };
 
+  nativeBuildInputs = [
+    faketty
+    nodejs
+    pnpm.configHook
+    makeBinaryWrapper
+  ];
+
+  # workaround for https://github.com/nrwl/nx/issues/22445
+  buildPhase = ''
+    runHook preBuild
+
+    faketty pnpm run bundle-for-release --disableRemoteCache=true --nxBail=true --outputStyle=static
```

Comment:
I usually pipe to cat

Reply:
It wants a TTY, piping to cat will not work. I have tried and it has failed.